src/coreclr/coreclr.proj

```diff
@@ -5,12 +5,12 @@
   <Target Name="Build">
     <ItemGroup>
       <_CoreClrBuildArg Condition="$([MSBuild]::IsOsPlatform(Windows))" Include="-skiptests" />
-      <_CoreClrBuildArg Include="-$(Platform)" />
+      <_CoreClrBuildArg Condition="'$(TargetArchitecture)' != ''" Include="-$(TargetArchitecture)" />
```

Comment:
Hmm, is there any valid scenario where we need to build CoreCLR without a target architecture specified?

Reply:
We don't need to specify the arch in local builds when we're building the same architecture as we're currently running on. I can likely clean this up to always calculate the argument correctly, but I'd have to take another look (this was an early fix to get everything building).

Reply:
Non-blocking from my side, I was just curious. I guess that if you're in the final round of testing, you can safely ignore that for now, but I doubt so considering the current infra state :-). In fact it would be probably better to tackle in a follow-up change one way or the other, the change is already daunting w.r.t. its complexity and additional churn won't help it unless absolutely necessary for fixing severe issues, at least that's my opinion.